### PR TITLE
Performance: 1.2% Avoid DSR extractIdFromUri

### PR DESCRIPTION
**Pull request by kirankumarkolli**
Optimized implementation avoids expensive StringUtils.Split

Before the FlameGraph is like with 0.4% CPU samples
image

Full benchmark run shows AVG of 1.2% total time gain.
> Before change 83, 82, 78, 81, 82 -> AVG = 81.2
> After change   82, 79, 80, 81, 79 -> AVG = 80.2

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentServiceRequest.java [hunk 1]
```diff
@@ -17,6 +17,7 @@
 import java.net.URI;
 import java.nio.ByteBuffer;
 import java.nio.charset.StandardCharsets;
+import java.security.InvalidParameterException;
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentServiceRequest.java:20]: unused import

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentServiceRequest.java [hunk 2]
```diff
@@ -713,40 +714,57 @@ public static RxDocumentServiceRequest createFromName(
         );
     }
 
+    // The path will be in the form of
+    // [/]{resourceType/resourceId}+[/] or
+    // [/]{resourceTyp]/resourceI]}+/resourceType[/]
+    //
+    // Expected output in both cases are [resourceId]
     private static String extractIdFromUri(String path) {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentServiceRequest.java:722]: Please consider adding unit tests for `RxDocumentServiceRequest#extractIdFromUri()`
or unit tests for `RxDocumentServiceRequest` which captures the relevant code change.

Comment (kirankumarkolli) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentServiceRequest.java]: Its actually dead code

**Pull request merged**